.github/workflows/ci.yaml

```diff
@@ -69,6 +74,22 @@ jobs:
           # as the placeholder for the auth token
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
+      - name: Use snapshot version
+        ## This step only runs for "snapshot" branches starting with "snapshot/"
+        ## We have to snapshot-version again in this job because the version
+        ## change in build-frontend only impacts the compiled assets. We need
+        ## the correct version again here for the publish to work properly.
```

Comment:
it took me a while to understand exactly what was the issue.
So I would suggest this comment instead: 

```
The previous job that is building the UI assets doesn't persist the different files `package.json` that has been modified by the script.
So in order to be able to publish the snapshot, we need to ensure the different files `package.json` contain the correct version. That's why we are running again the script to bump a snapshot version.

Reply:
Thanks for the rewording! I was struggling with how to communicate this well.